src/zarr/api/asynchronous.py

```diff
@@ -337,7 +337,10 @@ async def save_group(
     kwargs
         NumPy arrays with data to save.
     """
-    zarr_format = _handle_zarr_version_or_format(zarr_version=zarr_version, zarr_format=zarr_format)
+    zarr_format = (
+        _handle_zarr_version_or_format(zarr_version=zarr_version, zarr_format=zarr_format)
```

Comment:
Minor, but maybe `handle_zarr_version_or_format` could do the fallback to `_default_zarr_version()`. Then we can remove all the `or`s.

Reply:
I see what you are saying but we don't always do the fallback -- or in some cases, where we try to open without the fallback before creating with it.